streaming/src/main/scala/org/apache/spark/streaming/scheduler/StreamingSparkListenerAdapter.scala

```diff
@@ -17,19 +17,16 @@
 
 package org.apache.spark.streaming.scheduler
 
-import java.util.concurrent.atomic.AtomicBoolean
+import org.apache.spark.scheduler.{SparkListener, SparkListenerEvent}
 
-import org.apache.spark.Logging
-import org.apache.spark.util.AsynchronousListenerBus
-
-/** Asynchronously passes StreamingListenerEvents to registered StreamingListeners. */
-private[spark] class StreamingListenerBus
-  extends AsynchronousListenerBus[StreamingListener, StreamingListenerEvent]("StreamingListenerBus")
-  with Logging {
-
-  private val logDroppedEvent = new AtomicBoolean(false)
+/**
+ * A SparkListener adapter for StreamingListener that will dispatch the Streaming events to
+ * the underlying StreamingListener.
+ */
+private[spark] class StreamingSparkListenerAdapter(listener: StreamingListener)
```

Comment:
Making StreamingListener extend SparkListener has a tricky issue. 

Image someone implements both StreamingListener and SparkListener in the current codes. Then he needs to register this one with SparkContext and StreamingContext. If we make StreamingListener extend SparkListener, the user usually won't notice this change as his codes can be compiled successfully, however, the listener receives SparkListenerEvent twice and cause very tricky issue and hard to find the real problem.